### ISSUE TITLE
[AMRVAC] Optionally supply parfiles to populate AMRVACDataset.parameters

### ISSUE THREAD

**Issue by neutrinoceros** [yt-project/yt#2368]
## feature proposal

Populate `ds.parameters` with a `"namelist"` (working name) key, which value would be a `f90nml.Namelist` object, defaulting to `None`, through the optional load kwarg `parfiles`.

This optional argument should also be stored in `ds.parameters` to keep track of where information came from in a dataset.

**example user code**

```python
# minimal example
ds = yt.load("bw_2d0000.dat", parfiles="amrvac.par")

# a more complex example
parfiles = ["amrvac.par", "add_dust.par", "boundary_conditions.nml"]
ds = yt.load("bw_2d0000.dat", parfiles=parfiles)
```

**benefits**
- allow building of (user *or* default) derived fields using variables that are not found in the datfile header, such as pressure and sound-speed (and Mach number, while we're at it)
- more debugging information available to the user

**Requirements**
- on-demand-import for f90nml (this was already done here https://github.com/yt-project/yt/pull/2347)
- an additional routine to "patch" configurations on top of each other replicating AMRVAC's behaviour (I wrote exactly this in a personal codebase already)
- this needs to stay optional, although automatically discovering existing "amrvac.par" could help in many user cases (open to discussion)

**possible follow-up**
A subsequent feature proposal would be to expose the amrvac-namelist-parsing function so it can be used in a user-script to easily merge parfiles in one line

```python
from yt.frontends.amrvac import read_amrvac_namelist

parfiles = ["amrvac.par", "add_dust.par", "boundary_conditions.nml"]
unified_namelist = read_amrvac_namelist(parfiles)
unified_namelist.write("unique_namelist.par")
```
~I'll need help in decided where to put that function.~ (solved with help from @ngoldbaum)

ping @n-claes

**Comment by neutrinoceros** [comment]
companion PR https://github.com/yt-project/yt/pull/2369

**Comment by neutrinoceros** [comment]
fixed in #2369
